org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/cache/BasicCache.java

```diff
@@ -37,13 +38,14 @@
 public class BasicCache<K, V> implements ICache<K, V> {
 
   protected final String m_cacheId;
+  protected Supplier<String> m_labelSupplier;
   protected final ICacheValueResolver<K, V> m_resolver;
   protected final Map<K, V> m_cacheMap;
 
   protected final AbstractTransactionalMap<K, ?> m_transactionalMap; // is null if not transactional cache
 
-  public BasicCache(String cacheId, ICacheValueResolver<K, V> resolver, Map<K, V> cacheMap) {
-    this(cacheId, resolver, cacheMap, findTransactionalMap(cacheMap));
+  public BasicCache(String cacheId, Supplier<String> labelSupplier, ICacheValueResolver<K, V> resolver, Map<K, V> cacheMap) {
```

Comment:
You could keep the old constructor and mark it deprecated. This way, current users do not necessarily have to adapt the code.